test/fixtures_s3_test.go

```diff
@@ -0,0 +1,148 @@
+//+build integration fixtures
+
+package test_test
+
+import (
+	"fmt"
+	"github.com/applike/gosoline/pkg/blob"
+	"github.com/applike/gosoline/pkg/fixtures"
+	"github.com/applike/gosoline/pkg/mon"
+	"github.com/applike/gosoline/pkg/test"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"testing"
+)
+
+const (
+	BucketName    = "s3-fixtures-test-bucket"
+	BasePath      = "test_data/s3_fixtures_test_data"
+	BasePathPurge = "test_data/s3_fixtures_purge_test_data"
+)
+
+type FixturesS3Suite struct {
+	test.Suite
+
+	loader fixtures.FixtureLoader
+	client s3iface.S3API
+	logger mon.Logger
+	mocks  *test.Mocks
+}
+
+func (s *FixturesS3Suite) SetupSuite() []test.SuiteOption {
+	return []test.SuiteOption{
+		test.WithoutAutoDetectedComponents("localstack"),
+		test.WithLogLevel("debug"),
+		test.WithConfigFile("test_configs/config.s3.test.yml"),
+		test.WithConfigFile("test_configs/config.fixtures_s3.test.yml"),
+	}
+}
+
+func TestFixturesS3Suite(t *testing.T) {
+	s := new(FixturesS3Suite)
+	test.RunSuite(t, s)
+}
+
+func (s FixturesS3Suite) TestS3(app test.AppUnderTest) {
+	// CHECK: Alternative for the affectations below.
+	s.logger = mon.NewLogger()
+	s.loader = fixtures.NewFixtureLoader(s.Env().Config(), s.logger)
+	s.client = blob.ProvideS3Client(s.Env().Config())
+
+	err := s.loader.Load(s3DisabledPurgeFixtures())
+	s.NoError(err)
+
+	input := &s3.GetObjectInput{
+		Bucket: aws.String(BucketName),
+		Key:    aws.String(fmt.Sprintf("%s/nyan_cat2.gif", BasePath)),
+	}
+	output, err := s.client.GetObject(input)
+
+	s.NoError(err)
+	s.Equal(int64(28092), *output.ContentLength)
+
+	input = &s3.GetObjectInput{
+		Bucket: aws.String(BucketName),
+		Key:    aws.String(fmt.Sprintf("%s/subDir/nyan_cat1.gif", BasePath)),
+	}
+	output, err = s.client.GetObject(input)
+
+	s.NoError(err)
+	s.Equal(int64(28092), *output.ContentLength)
+
+	input = &s3.GetObjectInput{
+		Bucket: aws.String(BucketName),
+		Key:    aws.String(fmt.Sprintf("%s/subDir/nyan_cat.gif", BasePath)),
+	}
+	output, err = s.client.GetObject(input)
+
+	s.NoError(err)
+	s.Equal(int64(28092), *output.ContentLength)
```

Comment:
Since you are checking all files below a certain path, you could also use filepath.Walk here to assert the exitence of your fixtures. Maybe in combination with an expected amount of files that are supposed to be in that folder.

Reply:
If you check you'll find that I am doing that manually, checking ALL files and keys to check the wanted behaviour. Since it's a test, it's better this way. Why reproduce what it's doing already ? doesn't make sense to me.